Issues with TypeScript scripts using external libs after upgrade from 4.6.17 to 4.6.22

**Describe the bug**  
The working 4.6.17 config has:

*Additional npm modules*: `rxjs`, `rxjs/operators`
*Activate syntax help for these npm modules*: `rxjs`, `rxjs/operators`

No other settings that are non-default.

My scripts import `rxjs` types at the top like this:

```ts
import { concat, Observable, empty, of } from 'rxjs';
import { distinctUntilChanged, tap, share } from 'rxjs/operators';

// More code.
```

This works fine with 4.6.17.

After upgrading to 4.6.22

* The "Additional npm module" `rxjs/operators` throws an error while installing because there is no such package. But I need it somehow for 4.6.17.
* After removing all of `rxjs/operators` from the config the adapter starts OK, but my scripts can no longer load their imports.

```
2020-08-05 08:39:27.237 - error: javascript.0 (2279) script.js.Automation.Bind_Amp_To_TV: TypeScript compilation failed:
--
import { concat, Observable, empty, of } from 'rxjs';
^
ERROR: Cannot find module 'rxjs' or its corresponding type declarations.
import { distinctUntilChanged, tap, share } from 'rxjs/operators';
^
ERROR: Cannot find module 'rxjs/operators' or its corresponding type declarations.
```

**To Reproduce**  
Steps to reproduce the behavior:
1. Create a script as outlined above and confirm it starts on 4.6.17
1. Upgrade to 4.6.22

**Expected behavior**  
The upgrade should be transparent.

**Versions:**  
 - Adapter version: 4.6.22
 - JS-Controller version: 3.1.5
 - Node version: v12.18.3
 - Operating system: Docker on Raspbian

Just some thoughts, I'll have to look at this after my vacation:
`rxjs/...` is not a separate package, but a path inside `rxjs`. This means you cannot install it seperately.

This error might be because we're doing some magic to make TypeScript work inside ioBroker
```
import { distinctUntilChanged, tap, share } from 'rxjs/operators';
^
ERROR: Cannot find module 'rxjs/operators' or its corresponding type declarations.
```
I'm pretty sure this is because I haven't considered this kind of imports.

--- a/lib/typescriptTools.js
+++ b/lib/typescriptTools.js
@@ -80,11 +80,14 @@ function resolveTypings(pkg, wrapInDeclareModule) {
     if (!rootTypings) return {};
 
     const packageRoot = path.dirname(packageJsonPath);
+    const normalizeImportPath = filename => path.normalize(
+        `node_modules/${pkgIncludesTypings ? '' : '@types/'}${pkg}/${path.relative(packageRoot, filename)}`
+    ).replace(/\\/g, '/');
 
     const ret = {};
 
     // We need to look at `import/export ... from 'modulename'` and `/// <reference path='...' />`
-    const importDtsRegex = /(?:import|export) .+ from ["'](\.\/[^"']+)["']/g;
+    const importDtsRegex = /(?:import|export) .+ from ["'](\.+\/[^"']+)["']/g;
     const pathReferenceRegex = /\/\/\/ <reference path=["']([^"']+)["'] \/>/g;
     const matchAllImports = string => [
         ...matchAll(importDtsRegex, string),
@@ -99,6 +102,9 @@ function resolveTypings(pkg, wrapInDeclareModule) {
     // include package.json in typings, so TypeScript can look up the correct entry point
     const relativePath = `node_modules/${pkgIncludesTypings ? '' : '@types/'}${pkg}/package.json`.replace(/\\/g, '/');
     ret[relativePath] = JSON.stringify(packageJson);
+
+    // Used to test whether a .d.ts file already uses "declare module" or not
+    const declareModuleRegex = /^\s*declare module/gm;
 
     // recursively load all typings
     const definitionQueue = [rootTypings];
@@ -116,9 +122,9 @@ function resolveTypings(pkg, wrapInDeclareModule) {
             return {};
         }
         // We need to store the filename relative to the base dir
-        const relativePath = `node_modules/${pkgIncludesTypings ? '' : '@types/'}${pkg}/${path.relative(packageRoot, filename)}`.replace(/\\/g, '/');
+        const relativePath = normalizeImportPath(filename);
         // If necessary, wrap the root typings (only those!)
-        ret[relativePath] = !pkgIncludesTypings && wrapInDeclareModule && filename === rootTypings
+        ret[relativePath] = wrapInDeclareModule && filename === rootTypings && !declareModuleRegex.test(fileContent)
             ? `declare module "${pkg}" { ${fileContent} }`
             : fileContent;
         // If this file references another .d.ts file, we need to load that too
@@ -128,7 +134,7 @@ function resolveTypings(pkg, wrapInDeclareModule) {
             // find out the correct path of the file we want to import
             .map(normalizeDTSImport)
             // Find all libs we have not loaded yet
-            .filter(file => !(file in ret))
+            .filter(file => !(normalizeImportPath(file) in ret))
             .forEach(file => definitionQueue.push(file));
     }
     return ret;

--- a/main.js
+++ b/main.js
@@ -180,9 +180,24 @@ function loadTypeScriptDeclarations() {
     ) {
         const installedLibs = adapter.config.libraries.split(/[,;\s]+/).map(s => s.trim());
         const wantsTypings = adapter.config.libraryTypings.split(/[,;\s]+/).map(s => s.trim());
+        // Add all installed libraries the user has requested typings for to the list of packages
         for (const lib of installedLibs) {
             if (
                 wantsTypings.indexOf(lib) > -1
+                && packages.indexOf(lib) === -1
+            ) {
+                packages.push(lib);
+            }
+        }
+        // Some packages have sub-modules (e.g. rxjs/operators) that are not exposed through the main entry point
+        // If typings are requested for them, also add them if the base module is installed
+        for (const lib of wantsTypings) {
+            // Extract the package name and check if we need to add it
+            if (lib.indexOf('/') === -1) continue;
+            const pkgName = lib.substr(0, lib.indexOf('/'));
+
+            if (
+                installedLibs.indexOf(pkgName) > -1
                 && packages.indexOf(lib) === -1
             ) {
                 packages.push(lib);

